### PR TITLE
Add plugin versioning

### PR DESCRIPTION
**Pull request by leoparente**

### CODE REVIEW

File: src/CoreRegistry.cpp [hunk 1]
```diff
@@ -42,37 +42,38 @@ void CoreRegistry::start(HttpServer *svr)
...
                 if (_input_registry.loadState(s) == Corrade::PluginManager::LoadState::NotLoaded) {
                     _input_registry.load(s);
                 }
-                InputPluginPtr mod = _input_registry.instantiate(s);
-                _logger->info("Load input stream plugin: {} {}", s, mod->pluginInterface());
-                mod->init_plugin(this, svr);
-                _input_plugins.insert({s, std::move(mod)});
+                for (const auto &alias : meta->provides()) {
+                    InputPluginPtr mod = _input_registry.instantiate(alias);
+                    _logger->info("Load input stream plugin: {} {}", alias, mod->pluginInterface());
```
Comment (weyrick) [inline src/CoreRegistry.cpp]: let's add the version to this message too

File: src/CoreRegistry.cpp [hunk 2]
```diff
@@ -82,10 +83,19 @@ void CoreRegistry::start(HttpServer *svr)
...
-                _logger->info("Load stream handler plugin: {} {}", s, mod->pluginInterface());
-                mod->init_plugin(this, svr);
-                _handler_plugins.insert({s, std::move(mod)});
+                if (!meta->data().hasValue("version")) {
+                    _logger->error("version field is mandatory and was not provided by '{}'", s);
+                }
+                auto version = meta->data().value("version");
+                for (const auto &alias : meta->provides()) {
+                    HandlerPluginPtr mod = _handler_registry.instantiate(alias);
+                    _logger->info("Load stream handler plugin: {} {}", alias, mod->pluginInterface());
```
Comment (weyrick) [inline src/CoreRegistry.cpp]: version here too

**Review by weyrick — APPROVED** [review]

File: src/CoreRegistry.cpp (hunk 2 above)
Comment (leoparente) [reply src/CoreRegistry.cpp]: done

File: src/CoreRegistry.cpp (hunk 1 above)
Comment (leoparente) [reply src/CoreRegistry.cpp]: done

**Pull request merged**